platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/userlocation/MyLocationDrawableActivity.java

```diff
@@ -66,29 +67,29 @@ public void onMapReady(MapboxMap map) {
 
   @Override
   protected void enableLocation(boolean enabled) {
-    if (enabled) {
-      mapboxMap.setMyLocationEnabled(true);
-      Location location = mapboxMap.getMyLocation();
-      if (location != null) {
-        onLocationChanged(location);
-      } else {
-        LocationSource.getLocationEngine(this).addLocationEngineListener(this);
-      }
-    } else {
-      mapboxMap.setMyLocationEnabled(false);
+    mapboxMap.setMyLocationEnabled(enabled);
+    if (lostApiClient == null) {
+      lostApiClient = new LostApiClient.Builder(this).addConnectionCallbacks(this).build();
```

Comment:
Why are we using the `LostApiClient` directly here? Would it be better to encapsulate under a custom `LocationEngine`?